R/subset_datarqst.R

```diff
@@ -30,73 +42,189 @@
 
   }
 
-  message("Output subset as .csv files...")
+  #If keeping conflicted species: identify conflicted species individuals that were
+  # never identified as one of the species of interest and remove them from output
+  if(include_conflicting){
+
+    message("Identifying relevant conflicting-species individuals...")
+
+    #Set species of interest
+    if(!is.null(PopSpec_Filter)){
+      SpeciesInt <- unique(stringr::str_split(string = PopSpec_Filter_original, pattern = "_", simplify = TRUE)[,2])
+    }else{
+      SpeciesInt <- Species_original
+    }
+
+    #Make a list of all individuals with conflicting species
+    ConflSpec_IndvIDs <- output_individual %>%
+      dplyr::filter(.data$Species %in% c('CONFLICTED', 'CCCCCC')) %>%
+      dplyr::select(.data$IndvID)
+
+    #Progress bar
+    pb <- progress::progress_bar$new(total = length(unique(ConflSpec_IndvIDs$IndvID)),
+                                     format = "[:bar] :percent ~:eta remaining",
+                                     clear = FALSE)
 
-  save_path <- dirname(file)
+    #Determine which individuals were never identified as a species of interest
+    Irrelevant_IndvIDs <- output_capture %>%
+      dplyr::filter(.data$IndvID %in% ConflSpec_IndvIDs$IndvID) %>%
+      dplyr::arrange(.data$IndvID, .data$BreedingSeason, .data$CaptureDate, .data$CaptureTime) %>%
+      dplyr::group_by(.data$IndvID) %>%
+      dplyr::summarise(Relevant = purrr::map_chr(.x = list(unique(na.omit(.data$Species))),
+                                                 .f = ~{
+                                                 pb$tick()
+        if(any((..1) %in% SpeciesInt)){
 
-  utils::write.csv(x = output_brood, file = paste0(save_path, "\\Brood_data", as.numeric(Sys.Date()), ".csv"), row.names = F)
+          return("yes")
 
-  utils::write.csv(x = output_capture, file = paste0(save_path, "\\Capture_data", as.numeric(Sys.Date()), ".csv"), row.names = F)
+        } else {
 
-  utils::write.csv(x = output_individual, file = paste0(save_path, "\\Individual_data", as.numeric(Sys.Date()), ".csv"), row.names = F)
+          return("no")
 
-  utils::write.csv(x = output_location, file = paste0(save_path, "\\Location_data", as.numeric(Sys.Date()), ".csv"), row.names = F)
+        }
 
-  invisible(NULL)
+      }), .groups = "drop") %>%
+      dplyr::rowwise() %>%
+      #dplyr::ungroup() %>%
+      dplyr::filter(.data$Relevant == "no")
+
+    #Remove individuals never identified as one of the species of interest
+    output_individual <- output_individual %>%
+      filter(!(.data$IndvID %in% Irrelevant_IndvIDs$IndvID))
+
+    output_capture <- output_capture %>%
+      filter(!(.data$IndvID %in% Irrelevant_IndvIDs$IndvID))
+  }
+
+  #Combine output into one R object (and return it)
+  output_data <- list(Brood_data = output_brood,
+                      Capture_data = output_capture,
+                      Individual_data = output_individual,
+                      Location_data = output_location)
```

Comment:
When looking at the output of the subsetting function, I wondered whether we should re-assign Row in each table, so that it starts at 1 rather than the lowest row number the subset had in the .RDS of the standard data.

I also wondered whether we should order the tables by a date or year column before returning the subsetted data rather than the unique identifier of that table.

Reply:
@StefanVriend regarding the row number, this we need to discuss and I will send a mail to the dev team. 

As for the ordering of tables, I agree with you that it makes sense to order the tables in a more intuitive/natural way. 
I am now ordering tables as follows: 
Brood: PopID > BreedingSeason > Species > BroodID
Capture: CapturePopID > IndvID > BreedingSeason > CaptureDate > CaptureTime
Individual: PopID > Species > IndvID
Location: PopID > LocationType > StartSeason > EndSeason > LocationID

This ordering is what I personally think people may be most used to. Can easily be changed though if someone has strong opinions. 

(If we are going to re-assign row numbers, we should do that after ordering the tables I think).